cql3/expr/expression.cc

```diff
@@ -2567,5 +2567,68 @@ bool has_only_eq_binops(const expression& e) {
 
     return non_eq_binop == nullptr;
 }
+
+bool is_prepared(const expression& maybe_prepared) {
+    bool contains_unprepared = recurse_until(maybe_prepared, [](const expression& e) -> bool {
```

Comment:
I'm skeptical that we need it. We should always prepare everything early, and there's a limited number of places where we have unprepared expressions.

Reply:
I added the checks because once I forgot to prepare the expression before passing it and then I had to do some debugging to find out what''s wrong. A check like this would quickly show me what the error was. It's a bit defensive, but it clearly shows the type of arguments this code accepts. When working with expressions code I often feel uneasy because I'm not sure what type of arguments are valid, and which ones are not. Checks like this would make it easier for someone unfamiliar with the code to reason about things.
I can remove the checks if you think they're too much.